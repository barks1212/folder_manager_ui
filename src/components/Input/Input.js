import React from 'react';
import PropTypes from 'prop-types';

import {noop} from '../../utils/utils';
import classes from './Input.module.css';

const input = ({ filterFiles }) => (
    <div className={classes.InputWrapper}>
        <i className="fa fa-search" />
        <input placeholder='Filename...' onChange={filterFiles} />
    </div>
);

input.propTypes = {
    filterFiles: PropTypes.func
};

input.defaultProps = {
    filterFiles: noop
};

export default input;