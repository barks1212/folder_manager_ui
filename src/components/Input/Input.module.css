.InputWrapper {
    margin: 1rem;
}