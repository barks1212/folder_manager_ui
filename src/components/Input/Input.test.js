import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';

import Input from './Input';

describe('input', () => {

    it('should render', () => {
        const filterFilesMock = jest.fn();

        const {asFragment} = render(<Input filterFiles={filterFilesMock}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should have "Filename..." as a placeholder', () => {
        const filterFilesMock = jest.fn();

        const {container} = render(<Input filterFiles={filterFilesMock} />);
        const input = container.querySelector('input');

        expect(input.placeholder).toBe('Filename...');
    });

    it('should change value on user input', () => {
        const filterFilesMock = jest.fn();

        const {container} = render(<Input filterFiles={filterFilesMock} />);
        const input = container.querySelector('input');

        fireEvent.change(input, {target: {value: 'test'}});

        expect(input.value).toBe('test');
    });

    it('should call filter function on user input', () => {
        const filterFilesMock = jest.fn();

        const {container} = render(<Input filterFiles={filterFilesMock} />);
        const input = container.querySelector('input');

        fireEvent.change(input, {target: {value: 't'}});

        expect(filterFilesMock).toHaveBeenCalled();
    });
});