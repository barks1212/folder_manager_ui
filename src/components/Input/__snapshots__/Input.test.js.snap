// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input should render 1`] = `
<DocumentFragment>
  <div
    class="InputWrapper"
  >
    <i
      class="fa fa-search"
    />
    <input
      placeholder="Filename..."
    />
  </div>
</DocumentFragment>
`;
